browser/model/helpers/installer.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+let fs = require('fs-extra');
+let child_process = require('child_process');
+let unzip = require('unzip');
+
+import Logger from '../../services/logger';
+
+class Installer {
+
+  constructor(key, progress, success, failure) {
+    this.progress = progress;
+    this.success = success;
+    this.failure = failure;
+    this.key = key;
+  }
+
+  exec(command, options, result) {
+    let self = this;
+    return new Promise(function (resolve, reject) {
+      Logger.info(self.key + ' - Execute ' + command);
+      child_process.exec(command, options, function(error, stdout, stderr) {
```

Comment:
I'm not sure I follow you here. I'm using the let statement, because this.key is not accessible from within the closure function of the promise constructor.